### PR TITLE
Add weldable door sync

### PR DESCRIPTION
**Pull request by Coding-Hen**
Resolves second part of #1148

Added weldable panel syncing

Added ondoortoggle to sync doors opening/closing and persistance

Also futureproofs the locking and unlocking of the starship doors

### CODE REVIEW

File: NitroxClient/GameLogic/Spawning/Metadata/StarshipDoorMetadataProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using NitroxModel.Logger;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata
+{
+    class StarshipDoorMetadataProcessor : GenericEntityMetadataProcessor<StarshipDoorMetadata>
+    {
+        public override void ProcessMetadata(GameObject gameObject, StarshipDoorMetadata metadata)
+        {
+            Log.Info($"Received door metadata change for {gameObject.name} with data of {metadata}");
```
Comment (Jannify) [inline NitroxClient/GameLogic/Spawning/Metadata/StarshipDoorMetadataProcessor.cs:11]: I think that this (and all other lines in the other processors) is more like Debug

File: NitroxPatcher/Patches/Dynamic/StarshipDoor_OnDoorToggle_Patch.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    class StarshipDoor_OnDoorToggle_Patch : NitroxPatch, IDynamicPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(StarshipDoor);
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/StarshipDoor_OnDoorToggle_Patch.cs:14]: Minor: Merge L14 & 15 like in the LockDoor patch.

Comment (Coding-Hen) [reply NitroxPatcher/Patches/Dynamic/StarshipDoor_OnDoorToggle_Patch.cs:14]: I believe unless I am mistaken that this is not possible as that way of doing that is only possible on public methods

Comment (Jannify) [reply NitroxPatcher/Patches/Dynamic/StarshipDoor_OnDoorToggle_Patch.cs:14]: I mean
```
public static readonly MethodInfo TARGET_METHOD = typeof(StarshipDoor).GetMethod("OnDoorToggle", BindingFlags.Instance | BindingFlags.NonPublic); 

```

**Pull request merged**